packages/x-charts/src/BarChart/formatter.ts

```diff
@@ -57,6 +57,7 @@ const formatter: SeriesFormatter<'bar'> = (params, dataset) => {
       const dataKey = series[id].dataKey;
       completedSeries[id] = {
         layout: 'vertical',
+        labelMarkType: 'square',
```

Comment:
Moved markType default from `legend getter` to `series formatter` as the tooltip doesn't use the legend getter